ts/features/newProfile/store/reducers/index.ts

```diff
@@ -0,0 +1,279 @@
+      pot.map(profile, p => capitalize(`${p.name} ${p.family_name}`)),
+      undefined
+    )
+);
+
+// return true if the profile has an email
+export const hasProfileEmail = (user: InitializedProfile): boolean =>
+  user.email !== undefined;
+
+// return true if the profile has an email
+export const hasProfileEmailSelector = createSelector(
+  profileSelector,
+  (profile: NewProfileState): boolean =>
+    pot.getOrElse(
+      pot.map(profile, p => hasProfileEmail(p)),
+      false
+    )
+);
+
+// return the profile services preference mode
+export const profileServicePreferencesModeSelector = createSelector(
+  profileSelector,
+  (profile: NewProfileState): ServicesPreferencesModeEnum | undefined =>
+    pot.getOrElse(
+      pot.map(profile, p => p.service_preferences_settings.mode),
+      undefined
+    )
+);
+// return if the profile email user is already taken
+export const isProfileEmailAlreadyTakenSelector = createSelector(
+  profileSelector,
+  (profile: NewProfileState): boolean | undefined =>
+    pot.getOrElse(
+      pot.map(profile, p => p.is_email_already_taken),
+      undefined
+    )
+);
+
+// return true if the profile services preference mode is set (mode is set only when AUTO or MANUAL is the current mode)
+export const isServicesPreferenceModeSet = (
+  mode: ServicesPreferencesModeEnum | undefined
+): boolean =>
+  [ServicesPreferencesModeEnum.AUTO, ServicesPreferencesModeEnum.MANUAL].some(
+    sp => sp === mode
+  );
+
+// return true if the profile has an email and it is validated
+export const isProfileEmailValidated = (user: InitializedProfile): boolean =>
+  user.is_email_validated !== undefined && user.is_email_validated === true;
+
+// return true if the profile has an email and it is validated
+export const isProfileEmailAlreadyTaken = (user: InitializedProfile): boolean =>
+  !!user.is_email_already_taken;
+
+// Returns true if the profile has service_preferences_settings set to Legacy.
+// A profile that has completed onboarding will have this value mandatory set to auto or manual
+export const isProfileFirstOnBoarding = (user: InitializedProfile): boolean =>
+  user.service_preferences_settings.mode === ServicesPreferencesModeEnum.LEGACY;
+
+// Same as above, but Selector
+export const isProfileFirstOnBoardingSelector = createSelector(
+  profileSelector,
+  (profile: NewProfileState): boolean | undefined =>
+    pot.getOrElse(
+      pot.map(profile, p => isProfileFirstOnBoarding(p)),
+      undefined
+    )
+);
+
+// return true if the profile pot is some and its field is_email_validated exists and it's true
+export const isProfileEmailValidatedSelector = (state: GlobalState) =>
+  pipe(
+    state,
+    profileSelector,
+    profileStatusPot =>
+      pot.map(
+        profileStatusPot,
+        profileStatus =>
+          hasProfileEmail(profileStatus) &&
+          isProfileEmailValidated(profileStatus)
+      ),
+    profileEmailValidatedPot => pot.getOrElse(profileEmailValidatedPot, false)
+  );
```

Comment:
What do you think to remove this selector? Is usefull fo the practical exmaple? Same questions for others selectors